File: 프로그래머스/PCCE 기출문제/9번/이웃한 칸.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<string>> board, int h, int w) {
    int di[4] = {-1, 0, 1, 0};
    int dj[4] = {0, 1, 0, -1};
    
    int answer = 0;
    for(int i=0; i<4; i++){
        int tmp_h = h+di[i];
        int tmp_w = w+dj[i];
        
        if(tmp_h<0 || tmp_h>=board.size()) continue;
        if(tmp_w<0 || tmp_w>=board.size()) continue;
        
        if(board[tmp_h][tmp_w] == board[h][w]) answer++;
    }
    
    return answer;
}

File: 프로그래머스/PCCE 기출문제/9번/이웃한 칸.py
def solution(board, h, w):
    di = [-1, 0, 1, 0]
    dj = [0, 1, 0, -1]

    answer = 0
    for i in range(4):
        tmp_h = h+di[i]
        tmp_w = w+dj[i]

        if tmp_h<0 or tmp_h>=len(board): continue
        if tmp_w<0 or tmp_w>=len(board): continue

        if board[tmp_h][tmp_w] == board[h][w]: answer += 1

    return answer
